### PR TITLE
Adding Preset Transforms in reference scripts

### PR DESCRIPTION
**Pull request by datumbox**
Updating the following reference scripts:
- [x] Classification
- [x] Object Detection
- [x] Segmentation
- [x] Video Classification

The Similarity reference script was skipped because it's not a real recipe. A reference implementation for it can be seen at https://github.com/pytorch/vision/pull/3317/commits/71b70910dd472aa7d29ce6adebf1e6304a563b55.

### CODE REVIEW

**Review by fmassa — APPROVED** [review]
The proposed approach looks great to me, good to go for the other tasks as well!

**Review by fmassa — APPROVED** [review]
Changes for segmentation and video classification looks good to me as well!

File: references/classification/presets.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        self.transforms = transforms.Compose(trans)
+
+    def __call__(self, img):
+        return self.transforms(img)
+
+
+class ClassificationPresetEval:
+    def __init__(self, crop_size, resize_size=256, mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)):
+
+        self.transforms = transforms.Compose([
```
Comment (zhangguanheng66) [inline references/classification/presets.py:29]: Usually, for text domain, we will need to download the transform, for example sentencepiece model, or a vocabulary saved in text file.

Comment (datumbox) [reply references/classification/presets.py:29]: I'm writing here what we discussed on the call.

It seems that supporting your case is possible by using PyTorch Hub's `load_state_dict_from_url()` method and then passing the result to your code. This is very common pattern in TorchVision, used mainly for pre-trained models. Example:
https://github.com/pytorch/vision/blob/97885cb1c983525dd14b82f38760221cd4ea7a17/torchvision/models/mobilenetv3.py#L244-L249

**Comment by datumbox** [comment]
@netw0rkf10w Providing the training and inference transforms of each model/pipeline in an organized way so that people can reproduce the models. Here is an example of complex presets: [1](https://github.com/pytorch/vision/blob/master/references/detection/transforms.py) and [2](https://github.com/pytorch/vision/blob/master/references/detection/presets.py).

**Comment by datumbox** [comment]
Unfortunately many of these discussions happened outside of Github. This is quite problematic and we want to change it because in circumstances like this, it's hard to give information to people and does not help with the transparency... So, apologies for not being able to point you to a public thread... To solve the situation, below I provide a summary of what motivated the change. Let me know if you need more info:

Preset preprocessing transforms are those transformations applied to the data before feeding them to an ML model. They are typically separated into two categories: those applied during training and those during inference. Examples of such transforms include the Data Augmentation techniques, the Normalization/Scaling methods and other adhoc transformations  applied to the data as a preliminary step (binary to bitmap conversion for Vision, Fast Fourier Transforms for Audio, Tokenization for Text etc).

The preset transforms are a crucial part of the model and having access to them is necessary to understand how a model was created and how to use it. Disclosing which training transforms were used is an important part for reproducibility and crucial to understand the assumptions and properties a specific model. The latter is particularly true while using transfer learning and porting a model from one domain to another (for example the Zoom transform can be used for augmentation in ImageNet Classification but not for Cancer Detection). Similarly having access to the transforms used during inference is critical because without them one can’t use the model. 

This is the reason we decided to bring these Preset transforms as close to the training references as possible. By putting them together, the users are able to reproduce the training, adjust the scripts to meet their needs, do transfer learning etc. Hope that makes sense.

**Comment by netw0rkf10w** [comment]
@datumbox That totally makes sense! Thank you very much for your detailed explanation!
